x/nodes/keeper/valStateChanges_test.go

```diff
@@ -350,6 +352,129 @@ func TestValidatorStateChange_EditAndValidateStakeValidatorAfterNonCustodialUpgr
+	// Create and fund accounts
+	operatorAcc1 := getRandomPubKey()
+	operatorAcc2 := getRandomPubKey()
+	operatorAcc3 := getRandomPubKey()
+	operatorAddr1 := sdk.Address(operatorAcc1.Address())
+	operatorAddr2 := sdk.Address(operatorAcc2.Address())
+	operatorAddr3 := sdk.Address(operatorAcc3.Address())
+	assert.Nil(t, fundAccount(ctx, k, operatorAddr1, stakeAmount))
+	assert.Nil(t, fundAccount(ctx, k, operatorAddr2, stakeAmount))
+	assert.Nil(t, fundAccount(ctx, k, outputAddress, stakeAmount))
+
+	// Stake two nodes before NCUST
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = outputAddress
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc1))
+	msgTemplate.PublicKey = operatorAcc2
+	msgTemplate.Output = outputAddress
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc2))
+
+	// Verify staked nodes having nil output address
+	validatorCur, found := k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.Nil(t, validatorCur.OutputAddress)
+	validatorCur, found = k.GetValidator(ctx, operatorAddr2)
+	assert.True(t, found)
+	assert.Nil(t, validatorCur.OutputAddress)
+
+	// Enable NCUST
+	codec.UpgradeFeatureMap[codec.NonCustodialUpdateKey] = -1
+
+	// Set an output address on Addr1 after NCUST
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = outputAddress
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc1))
+	validatorCur, found = k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.True(t, validatorCur.OutputAddress.Equals(outputAddress))
+
+	// Attempt to change the output address with operator's signature --> Fail
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err := runStake(ctx, k, msgTemplate, operatorAcc1)
+	assert.Equal(t, k.codespace, err.Codespace())
+	assert.Equal(t, types.CodeUnequalOutputAddr, err.Code())
+
+	// Attempt to change the output address with output's signature --> Fail
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err = runStake(ctx, k, msgTemplate, outputAccount)
+	assert.Equal(t, k.codespace, err.Codespace())
+	assert.Equal(t, types.CodeUnauthorizedSigner, err.Code())
+
+	// Enable OEDIT
+	codec.UpgradeFeatureMap[codec.OutputAddressEditKey] = -1
+
+	// Attempt to change the output address with operator's signature --> Fail
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err = runStake(ctx, k, msgTemplate, operatorAcc1)
+	assert.Equal(t, k.codespace, err.Codespace())
+	assert.Equal(t, types.CodeDisallowedOutputAddressChange, err.Code())
+
+	// Attempt to change the output address with output's signature --> Success
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err = runStake(ctx, k, msgTemplate, outputAccount)
+	assert.Nil(t, err)
+	validatorCur, found = k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.True(t, validatorCur.OutputAddress.Equals(operatorAddr1))
+
+	// Attempt to change the output address from nil
```

Comment:
This change is reasonable but wasn't clear from the github description. Please update it appropriately.